datatypes/src/operations/image/colorizer.rs

```diff
@@ -3,44 +3,140 @@ use crate::operations::image::RgbaTransmutable;
+            match json_key {
+                "type" => colorizer_type = map.next_value()?,
+                "breakpoints" => breakpoints = map.next_value()?,
+                "colors" => colors = map.next_value()?,
+                "noDataColor" => no_data_color = map.next_value()?,
+                "overColor" => over_color = map.next_value()?,
+                "underColor" => under_color = map.next_value()?,
+                "defaultColor" => {
+                    // broadcast the legacy value to both new values
+                    let default_color: Option<RgbaColor> = map.next_value()?;
+                    over_color = default_color;
+                    under_color = default_color;
+                }
+                // TODO: should this throw an error? What if the json contains a redundant field,
+                // but is otherwise ok?
+                _ => (),
+            };
+        }
+
+        match colorizer_type {
+            Some("linearGradient") => Ok(Colorizer::LinearGradient {
+                breakpoints: breakpoints
+                    .ok_or_else(|| serde::de::Error::custom("Missing breakpoints field"))?,
+                no_data_color: no_data_color
+                    .ok_or_else(|| serde::de::Error::custom("Missing noDataColor field"))?,
+                over_color: over_color
+                    .ok_or_else(|| serde::de::Error::custom("Missing overColor field"))?,
+                under_color: under_color
+                    .ok_or_else(|| serde::de::Error::custom("Missing underColor field"))?,
+            }),
+            Some("logarithmicGradient") => Ok(Colorizer::LogarithmicGradient {
+                breakpoints: breakpoints
+                    .ok_or_else(|| serde::de::Error::custom("Missing breakpoints field"))?,
+                no_data_color: no_data_color
+                    .ok_or_else(|| serde::de::Error::custom("Missing noDataColor field"))?,
+                over_color: over_color
+                    .ok_or_else(|| serde::de::Error::custom("Missing overColor field"))?,
+                under_color: under_color
+                    .ok_or_else(|| serde::de::Error::custom("Missing underColor field"))?,
+            }),
+            Some("palette") => Ok(Colorizer::Palette {
+                colors: colors.ok_or_else(|| serde::de::Error::custom("Missing colors field"))?,
+                no_data_color: no_data_color
+                    .ok_or_else(|| serde::de::Error::custom("Missing noDataColor field"))?,
+                over_color: over_color
+                    .ok_or_else(|| serde::de::Error::custom("Missing overColor field"))?,
+                under_color: under_color
+                    .ok_or_else(|| serde::de::Error::custom("Missing underColor field"))?,
+            }),
+            //TODO: what about the `Rgba` variant?
+            _ => Err(serde::de::Error::custom(
+                "Encountered an unknown colorizer type.",
+            )),
+        }
+    }
+}
```

Comment:
Now, we cannot deserialize the Rgba variant anymore.

You could make use of serde's `#[serde(from = "FromType")]`, cf. https://serde.rs/container-attrs.html#from.

Then, you could have all fields on the deserializer structs, but optionally.
Then, you only succeed if you have either over/under OR default.
For Rgba, you just pass the enum variant through.

Then, we can say that this Deserializer struct will be removed in the future, removing the default_color option.
And this seems less brittle but more verbose than the thing above.

Reply:
Ok I've implemented it now according to your example from slack.
https://github.com/geo-engine/geoengine/pull/684/commits/d6c04ea287ef8ed52afd716ccdb7ca14e51aacf5